sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/ConfigurableProvider.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.spi;
+
+/**
+ * A named configurable provider.
+ *
+ * <p>It can be used to generically determine if a provider should be replaced by another provider
+ * with the same name.
+ */
+public interface ConfigurableProvider {
```

Comment:
Can we get rid of this interface? It makes this PR seem a lot more complicated that it is because it bundles together the introduction of `ComponentLoader` (the essence of the PR), with this refactoring to introduce a common interface for named providers.

Reply:
Yes, I figured we could just use the convention that the last occurrence of a "named" SPI wins - as it is currently implemented.